contributors/design-proposals/api-machinery/dynamic-admission-control-configuration.md

```diff
@@ -273,6 +282,23 @@ crashes, so it is rare.
 
 See [Considered but REJECTED alternatives](#considered-but-rejected-alternatives) for considered alternatives.
 
+## Enforce initializers to specific namespaces
+
+Current `InitializerConfiguration` is at the cluster level and all of the to-be-created resources (such as rc and deployments) defined in `Rules`
+will be appended with the pending initializers automatically during creating, regardless of the namespace.
+There is no way to only apply the initializers to specific namespaces.
+
+For example, when running a multi-tenant cluster, it'd be quite useful to only apply the rules in just certain namespaces. Sometimes we
+don't want to enforce in "kube-*" related namespaces as well.
+
+With the help of `NamespaceSelector`, we can
+
+* Apply initializer to ALL namespaces (by default);
+* Apply initializer to limited namespaces using label selector;
+
+Since most users won't add extra labels for namespaces explicitly when creating new resources, the selector matching should only be applied to
+`labels.Set(map[string]string{"namespace": namespace})` instead of widely-used `metadata.Labels`.
```

Comment:
is this proposing creating an artificial label set, disconnected from the actual namespace labels?

Reply:
Yes. Get`namespace` value from `attributeRecord` when admitting, `metadata.Labels` is not used/referenced here.

Reply:
@dixudx sorry, do not quite catch this, can you please show an example here? How to handle the case if a namespaces do not have labels?

Reply:
@gyliu513 Please refer to [findInitializers of PR #53879](https://github.com/dixudx/kubernetes/blob/7cb9ee46f9460f163cd23e2b1490a4f54ab60be8/staging/src/k8s.io/apiserver/pkg/admission/plugin/initialization/initialization.go#L298-L320) for the matching implementation.

And the `namespace` here is retrieved from `admission.Attributes` when admitting initializers. So `metadata.Labels` is not used.

Reply:
Got it, thanks @dixudx

Reply:
I'll bring this idea up when discussing the bootstrap problem of webhooks. I want to avoid artificial label set if possible.